plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -408,6 +417,125 @@ public void updatePartitionStatistics(HiveIdentity identity, Table table, String
+    {
+        ImmutableList.Builder<BatchUpdatePartitionRequestEntry> updateRequests = ImmutableList.builder();
+        ImmutableSet.Builder<GlueColumnStatisticsProvider.StatisticsUpdate> statisticsUpdates = ImmutableSet.builder();
+
+        Map<String, UpdatePartitionAsyncHandler> asyncHandlers = updates.stream().collect(
+                toImmutableMap(
+                        Entry::getKey,
+                        entry -> new UpdatePartitionAsyncHandler(table, entry.getValue())));
+        Map<String, Future<GetPartitionResult>> getPartitionResults = updates.stream().collect(
+                toImmutableMap(
+                        Entry::getKey,
+                        entry -> glueClient.getPartitionAsync(new GetPartitionRequest()
+                                        .withCatalogId(catalogId)
+                                        .withDatabaseName(table.getDatabaseName())
+                                        .withTableName(table.getTableName())
+                                        .withPartitionValues(toPartitionValues(entry.getKey())),
+                                asyncHandlers.get(entry.getKey()))));
+
+        getPartitionResults.forEach((partitionValues, getPartitionResult) -> {
+            getFutureValue(getPartitionResult);
+
+            UpdatePartitionAsyncHandler asyncHandler = asyncHandlers.get(partitionValues);
+            verify(asyncHandler.isFinished());
+
+            if (asyncHandler.getException().isPresent()) {
+                throw new TrinoException(HIVE_METASTORE_ERROR, asyncHandler.getException().get());
+            }
+
+            PartitionInput partitionInput = GlueInputConverter.convertPartition(asyncHandler.getPartition());
+            partitionInput.setParameters(asyncHandler.getPartition().getParameters());
+
+            updateRequests.add(new BatchUpdatePartitionRequestEntry()
+                    .withPartitionValueList(asyncHandler.getPartition().getValues())
+                    .withPartitionInput(partitionInput));
+            statisticsUpdates.add(new GlueColumnStatisticsProvider.StatisticsUpdate(asyncHandler.getPartition(), asyncHandler.getUpdatedColumnStatistics()));
+        });
+
+        Future<BatchUpdatePartitionResult> updatePartitionResult = glueClient.batchUpdatePartitionAsync(new BatchUpdatePartitionRequest()
+                .withCatalogId(catalogId)
+                .withDatabaseName(table.getDatabaseName())
+                .withTableName(table.getTableName())
+                .withEntries(updateRequests.build()));
+        try {
+            columnStatisticsProvider.updatePartitionStatistics(statisticsUpdates.build());
+            getFutureValue(updatePartitionResult);
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    private class UpdatePartitionAsyncHandler
```

Comment:
Sad that the sync and async API's of Glue have differences in features needing to do this additional work. Do newer versions also not support fetching partitions in bulk using the sync client?

Reply:
So you can get the partitions in bulk, but doing it this way also allowed me to run the PartitionStatistics update on an async thread, but maybe that's a hacky way of doing it. 

The problem is that in order to update the statistics we're first calling `getPartitionStatistics`, which is yet another API call. I should probably update this to allow for reading a set of Partitions at a time in GlueColumnStatisticsProvider and do the synchronization there.